src/contract.rs

```diff
@@ -80,27 +87,33 @@ pub mod execute {
             fallback: false,
         };
 
-        let mut tokens: Vec<Token> = vec![];
+        let mut tokens_id: Vec<Token> = vec![];
+        let mut tokens_take_profit: Vec<Token> = vec![];
         let retry_delay: u64 = RETRY_DELAY.load(deps.storage)?;
-        for deposit_id in deposit_ids {
+        for deposit in deposits {
+            let deposit_id = deposit.deposit_id;
+            let profit_taking = deposit.profit_taking_or_stop_loss;
             if let Some(timestamp) = WITHDRAW_TIMESTAMP.may_load(deps.storage, deposit_id)? {
                 if timestamp.plus_seconds(retry_delay).lt(&env.block.time) {
```

Comment:
This is the same bug as indicated in the audit PR's, but it's not part of this PR I guess.

Reply:
Bug? What's the bug exactly? @measure-fi